app/models/service_request.rb

```diff
@@ -344,22 +344,19 @@ def has_per_patient_per_visit_services?
 
   def total_direct_costs_per_patient arms=self.arms, line_items=nil
     total = 0.0
-    lids = line_items.map(&:id) unless line_items.nil?
     arms.each do |arm|
```

Comment:
Refactor these methods and replace slower enumerable methods with queries on eager_loaded line_items_visits, then eager load more data that's used in the cost totals